velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -951,4 +951,59 @@ struct MillisToTimestampFunction {
   }
 };
 
+template <typename T>
+struct TimestampDiffFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* unitString,
+      const arg_type<Timestamp>* /*timestamp1*/,
+      const arg_type<Timestamp>* /*timestamp2*/) {
+    if (unitString != nullptr) {
+      unit_ = fromDateTimeUnitString(
+          *unitString, /*throwIfInvalid=*/false, /*allowMicro=*/true);
```

Comment:
Should we throw an exception if the unit is invalid?

Reply:
https://github.com/apache/spark/blob/ebf5a2aef403e14906ecc9a9e56863bad957942e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L787

Reply:
Thanks, If the unit is illegal, Spark will report an error during analyze. 
Therefore, it is theoretically unreachable, but it is more suitable to keep it consistent, updated.